pkg/policy/selector/selector.go

```diff
@@ -22,6 +23,10 @@ type Selector []segment
 //   - a resolutionerr error if not being able to resolve to a node
 //   - nil and no errors, if the selector couldn't match on an optional segment (with ?).
 func (s Selector) Select(subject ipld.Node) (ipld.Node, error) {
+	if err := limits.ValidateIntegerBoundsIPLD(subject); err != nil {
```

Comment:
While that works, I don't think it gives enough guarantee. Here you check at "execution" time, but arguably a user should be prevented from making a token with args out of spec. This means that you should check the limits when creating args, or parsing args.

Reply:
@MichaelMure I moved it to the args "creation" (when they're added). https://github.com/ucan-wg/go-ucan/pull/89/commits/56eab758ed8b2b21dc902babf11c635d01fadd3b

where are they "parsed" in that sense?